source/IAccessibleHandler/orderedWinEventLimiter.py

```diff
@@ -89,10 +91,11 @@ def flushEvents(self):
 		threadCounters = {}
 		for k, v in sorted(g.items(), key=lambda item: item[1], reverse=True):
 			threadCount = threadCounters.get(k[-1], 0)
-			if threadCount > MAX_WINEVENTS_PER_THREAD:
-				continue
-			heapq.heappush(self._eventHeap, (v,) + k)
 			threadCounters[k[-1]] = threadCount + 1
+			if not alwaysAllowedObjects or k[1:-1] not in alwaysAllowedObjects:
+				if threadCount > MAX_WINEVENTS_PER_THREAD:
+					continue

```

Comment:
I think it would clarify the intent here to save the first condition with a name eg `eventsForObjectAlwaysAllowed` and combine them.

Then the logic can be in the positive, and it's there it is more clear there is only one scope for these two conditions.

```suggestion
			eventsForObjectAlwaysAllowed = alwaysAllowedObjects and k[1:-1] in alwaysAllowedObjects
			if threadCount > MAX_WINEVENTS_PER_THREAD and not eventsForObjectAlwaysAllowed :
				continue

```